### PR TITLE
cmd: add node list subcommand to list hubble nodes with status

### PR DESCRIPTION
**Pull request by Rolinh**
```
$ hubble nodes list
NAME                 STATUS      AGE      FLOWS/S   CURRENT/MAX-FLOWS
kind-control-plane   Connected   40m18s   1.03      2485/8191 ( 30.34%)
kind-worker          Connected   40m19s   6.75      8191/8191 (100.00%)
kind-worker2         Connected   40m18s   1.03      2479/8191 ( 30.26%)
kind-worker3         Connected   40m18s   1.03      2487/8191 ( 30.36%)
```

```
$ hubble nodes list -o wide
NAME                 STATUS      AGE      FLOWS/S   CURRENT/MAX-FLOWS     VERSION                     ADDRESS           TLS
kind-control-plane   Connected   40m48s   1.03      2510/8191 ( 30.64%)   cilium v1.9.90+ga8cef4401   172.18.0.2:4244   Enabled
kind-worker          Connected   40m49s   6.75      8191/8191 (100.00%)   cilium v1.9.90+ga8cef4401   172.18.0.4:4244   Enabled
kind-worker2         Connected   40m48s   1.02      2504/8191 ( 30.57%)   cilium v1.9.90+ga8cef4401   172.18.0.5:4244   Enabled
kind-worker3         Connected   40m48s   1.03      2512/8191 ( 30.67%)   cilium v1.9.90+ga8cef4401   172.18.0.3:4244   Enabled
```

```
$ hubble nodes list -o json
[
  {
    "name": "kind-control-plane",
    "version": "cilium v1.9.90+ga8cef4401",
    "address": "172.18.0.2:4244",
    "state": "NODE_CONNECTED",
    "tls": {
      "enabled": true,
      "server_name": "kind-control-plane.default.hubble-grpc.cilium.io"
    },
    "uptime_ns": "2477731396873",
    "num_flows": "2546",
    "max_flows": "8191",
    "seen_flows": "2546"
  },
  {
    "name": "kind-worker",
    "version": "cilium v1.9.90+ga8cef4401",
    "address": "172.18.0.4:4244",
    "state": "NODE_CONNECTED",
    "tls": {
      "enabled": true,
      "server_name": "kind-worker.default.hubble-grpc.cilium.io"
    },
    "uptime_ns": "2478761318015",
    "num_flows": "8191",
    "max_flows": "8191",
    "seen_flows": "16734"
  },
  {
    "name": "kind-worker2",
    "version": "cilium v1.9.90+ga8cef4401",
    "address": "172.18.0.5:4244",
    "state": "NODE_CONNECTED",
    "tls": {
      "enabled": true,
      "server_name": "kind-worker2.default.hubble-grpc.cilium.io"
    },
    "uptime_ns": "2477746517625",
    "num_flows": "2540",
    "max_flows": "8191",
    "seen_flows": "2540"
  },
  {
    "name": "kind-worker3",
    "version": "cilium v1.9.90+ga8cef4401",
    "address": "172.18.0.3:4244",
    "state": "NODE_CONNECTED",
    "tls": {
      "enabled": true,
      "server_name": "kind-worker3.default.hubble-grpc.cilium.io"
    },
    "uptime_ns": "2477735848470",
    "num_flows": "2548",
    "max_flows": "8191",
    "seen_flows": "2548"
  }
]
```

Note: depends on/blocked by https://github.com/cilium/cilium/pull/13979

### CODE REVIEW

File: cmd/node/list.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+			defer hubbleConn.Close()
+			return runList(ctx, cmd, hubbleConn)
+		},
+	}
+
+	cmd.Flags().StringVarP(
+		&listOpts.output, "output", "o", "table",
+		`Specify the output format, one of:
+ json:     JSON encoding
+ table:    Tab-aligned columns`)
```
Comment (glibsm) [inline cmd/node/list.go:59]: `wide` is missing from the list

File: cmd/node/list.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	case "table", "wide":
+		tw := tabwriter.NewWriter(cmd.OutOrStdout(), 2, 0, 3, ' ', 0)
+		fmt.Fprint(tw, "NAME\tSTATUS\tAGE\tFLOWS/S\tCURRENT/MAX-FLOWS")
+		if listOpts.output == "wide" {
+			fmt.Fprint(tw, "\tVERSION\tADDRESS\tTLS")
+		}
+		fmt.Fprintln(tw)
+
+		for _, n := range nodes {
+			age := "N/A"
```
Comment (glibsm) [inline cmd/node/list.go:98]: const the `N/A` maybe?

File: cmd/node/list.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+			}
+			flowsRatio := "N/A"
+			if maxFlows := n.GetMaxFlows(); maxFlows > 0 {
+				flowsRatio = fmt.Sprintf("%d/%d (%6.2f%%)", n.GetNumFlows(), maxFlows, (float64(n.GetNumFlows())/float64(maxFlows))*100)
+			}
+			version := "N/A"
+			if v := n.GetVersion(); v != "" {
+				version = v
+			}
+			fmt.Fprintf(
```
Comment (glibsm) [inline cmd/node/list.go:112]: afaict there is no formatting going on here. In regular hubble printer we just write with `n.GetName(), tab, age, tab, etc...`

File: cmd/node/list.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+			}
+			fmt.Fprintln(tw)
+		}
+		return tw.Flush()
+	default:
+		return fmt.Errorf("unknown output format: %s", listOpts.output)
+	}
+}
+
+func nodeStateToString(state relaypb.NodeState) string {
```
Comment (glibsm) [inline cmd/node/list.go:134]: i think we may want to upcase these values

File: cmd/node/node.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/spf13/viper"
+)
+
+// New creates a new hidden peer command.
+func New(vp *viper.Viper) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "nodes",
+		Aliases: []string{"node"},
+		Short:   "Get information about Hubble nodes",
+		Long:    `Get information about Hubble nodes.`,
```
Comment (glibsm) [inline cmd/node/node.go:28]: we can omit long if there is no additional information

File: cmd/node/node.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+package node
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// New creates a new hidden peer command.
+func New(vp *viper.Viper) *cobra.Command {
```
Comment (glibsm) [inline cmd/node/node.go:23]: You seem to have taken to this pattern of accepting viper into command constructor. What in particular does it make easier, no need for globals to float around?

File: cmd/node/list.go (hunk 4 above)
Comment (Rolinh) [reply cmd/node/list.go:134]: This is done with `strings.Title` at call site. Or do you mean we should title case the values here?

File: cmd/node/node.go (hunk 6 above)
Comment (Rolinh) [reply cmd/node/node.go:23]: Yes, no need for globals to float around and it eventually should also make implementing tests easier.

File: cmd/node/list.go (hunk 1 above)
Comment (Rolinh) [reply cmd/node/list.go]: Nice catch!

**Review by kAworu — APPROVED** [review]
:+1:

File: cmd/node/list.go (hunk 4 above)
Comment (glibsm) [reply cmd/node/list.go:133]: Yeah I was suggesting that we display (or always represent) the enum values in all-caps. Not super important

**Review by glibsm — APPROVED** [review]

**Pull request merged**